pkg/discovery/dtofactory/container_dto_builder.go

```diff
@@ -220,3 +241,14 @@ func (builder *containerDTOBuilder) addPodProperties(pod *api.Pod, index int) []
 
 	return properties
 }
+
+// Get the stitching property for Application.
+func getContainerStitchingProperty(pod *api.Pod, index int) string {
+	// Parse the container id from the status, it is either in the form of
+	// containerID: docker://986c8fc7247d1ff047f06e8e3487029b89baef1e908d3e334dd94aaa3bf4bc39
+	// or
+	// containerID: cri-o://81ab6978aa3252bc53cfaf7d23b8b1bf5f8f27ff4b9e79a0fbaffebeddaeeb4b
+	containerStatus := &(pod.Status.ContainerStatuses[index])
```

Comment:
Perhaps we can add a check here to ensure `ContainerStatuses` exists and it doesn't blow up, e.g.
```
if len(pod.Status.ContainerStatuses) == 0 {
  return ""
}
```
Or maybe even better this way?
```
if len(pod.Status.ContainerStatuses) <= index {
  return ""
}
```

Reply:
thanks